improver/nbhood/square_kernel.py

```diff
@@ -93,367 +95,58 @@ def __repr__(self):
-            ymax_xmin_disp,
-            n_rows,
-            n_columns,
-        )
-
-        if self.sum_or_fraction == "fraction":
-            # Initialise and calculate the neighbourhood area.
-            neighbourhood_area = self.calculate_neighbourhood(
-                summed_mask,
-                ymax_xmax_disp,
-                ymin_xmax_disp,
-                ymin_xmin_disp,
-                ymax_xmin_disp,
-                n_rows,
-                n_columns,
-            )
-
-            with np.errstate(invalid="ignore", divide="ignore"):
-                if iscomplex:
-                    cube.data = neighbourhood_total.astype(
-                        complex
-                    ) / neighbourhood_area.astype(complex)
-                else:
-                    cube.data = neighbourhood_total.astype(
-                        float
-                    ) / neighbourhood_area.astype(float)
-                cube.data[~np.isfinite(cube.data)] = np.nan
-        elif self.sum_or_fraction == "sum":
-            if iscomplex:
-                cube.data = neighbourhood_total.astype(complex)
-            else:
-                cube.data = neighbourhood_total.astype(float)
-
-        return cube
-
-    @staticmethod
-    def set_up_cubes_to_be_neighbourhooded(cube, mask_cube=None):
-        """
-        Set up a cube ready for neighourhooding the data.
-
-        Args:
-            cube (iris.cube.Cube):
-                Cube that will be checked for whether the data is masked
-                or nan. The cube should contain only x and y dimensions,
-                so will generally be a slice of a cube.
-            mask_cube (iris.cube.Cube):
-                Input Cube containing the array to be used as a mask.
-
-        Returns:
-            (tuple): tuple containing:
-                **cube** (iris.cube.Cube):
-                    Cube with masked or NaN values set to 0.0
-                **mask** (iris.cube.Cube):
-                    Cube with masked or NaN values set to 0.0
-                **nan_array** (numpy.ndarray):
-                    numpy array to be used to set the values within
-                    the data of the output cube to be NaN.
-
-        """
-        # Set up mask_cube
-        if not mask_cube:
-            mask = cube.copy()
-            mask.data = np.real(np.ones_like(mask.data))
+            area_mask = np.array(mask, dtype=area_mask_dtype, copy=False)
+        # Data mask to be eventually used for re-masking.
+        # (This is OK even if mask is None, it gives a scalar False mask then.)
+        data_mask = mask == 0
+        if isinstance(data, np.ma.MaskedArray):
+            # Include data mask if masked array
+            data_mask = data_mask | data.mask
+            data = data.data
+        if issubclass(data.dtype.type, np.complexfloating):
+            data_dtype = np.complex128
+        elif name.startswith("probability_of"):
+            data_dtype = np.float32
+            cumsum = True
```

Comment:
Temporary thing to keep a few acceptance tests originally done in float32 still working until they their KGOs get replaced.